### DISCUSSION TITLE
Can you debug Audacity with Visual Studio?

### DISCUSSION THREAD

**Discussion by werame** [audacity/audacity#2439] [Development]
If I run Audacity (RelWithDebInfo) in the VS debugger it just hangs at the little spalsh window

![image](https://user-images.githubusercontent.com/97036286/149681935-34768bd6-6d93-4b78-83d1-942261147ce0.png)

For like 5 minutes, but then opens its main window ok.

It does run further/normally outside the debugger. So, how do you debug it on Windows? Waiting 5 minutes every times seem a bit ridiculous. There's no CPU load or anything.

I could move development to Linux, but I'm assuming someone had to debug it on Windows.

**Comment by crsib** [comment]
MSVC is a daily driver for me and a few other devs. There is no issue with debugging. The possible exception is when you have symbol server enabled. In this case - the first launch can be slow while the symbols required are downloaded.

Debugging usually start within a few seconds, depending on ASIO and FFmpeg settings: 

https://user-images.githubusercontent.com/2660628/149744101-373f084f-e46b-4e23-93d0-9e0b17e7ad65.mp4

**Comment by werame** [comment]
You're right, it started much faster on a 2nd run. There is one thing that was slow inside Audacity itself though, in debug mode: the file-save dialog. Might again have been just an issue of first use, we'll see... Yeah, even the file-open is faster on a 3rd debug run. More symbols having been pulled already, I'm sure.
